### ISSUE TITLE
[BUG] Custom Headers are not included in requests/api calls.

### ISSUE THREAD

**Issue by MerelSollie** [LeoKlaus/Paperparrot#192]
**Describe the bug**
Custom Headers are not included with the /api/token call and possibly other requests on iOS 17.3.1 with Paperparrot.

**To Reproduce**
1. Go to '"Settings"
2. Click on 'Connect to a paperless-ng(x) server'.
3. Confirm the removal of all local documents.
4. Click 'Add custom headers'
5. Click '+ Add'.
6. Enter header name.
7. Enter header value.
8. Click 'Save'
9. Enter server URL.
10. Enter username and password.
11. Click 'Login'.
12. See error "Server responded: 403", which is expected behaviour when the authentication headers are not send to the Reverse Proxy.

**Steps taken to isolate issue:**
1. Switched to HTTP to MITM requests.
2. Enabled Wireguard on Reverse proxy machine, so the request can be intercepted/logged before it reaches the reverse proxy.
3. Send request.

The following headers were present:
```
Host: <removed for privacy reasons>
Content-Type: application/json
Connection: keep-alive
Accept: */*
User-Agent: Paperparrot/323 CFNetwork/1492.0.1 Darwin/23.3.0
Content-Length: 49
Accept-Language: nl-NL,nl;q=0.9
Accept-Encoding: gzip, deflate
```

The following headers were setup in Paperparrot and were not included by paperparrot in the request:
```
TestHeader: testvalue
X-Client-Id: <removed for security reasons>
X-Client-Secret: <removed for privacy reasons>
```

**Expected behavior**
It is expected that when custom headers are setup, they are included in every request, including the /api/token request.

**Screenshots**

**Environment(please complete the following information):**
 - Device: iPhone 11
 - OS: iOS 17.3.1
 - Version 2.2.0
 - Server Version (if applicable) N/A, as requests are not correctly send by the client.

**Additional context**
I initially discovered the issue when attempting to setup Cloudflare Zero Access Service authentication headers. This lead me to hosting a reverse proxy on my local machine, so I could intercept the request. 

This is my first use of the app, but I first believed the issue was on my end, and Service authentication headers are preferred to the implementation other apps provide, for me at least. 
I have attempted multiple re-installs of the app, and a device reboot. 
I have tried multiple different formats (e.g. quotes, escaping characters, only providing lowercase) and combinations of values in the "Custom headers" menu, and all have led to the same conclusion that the issue seems to be caused by the app simply not sending the headers.

The logs of one example request from the app's side:
```
[22/2/2024, 19:55] [ApiHandler] Connecting to <private>
[22/2/2024, 19:55] [ApiHandler] Obtaining token...
[22/2/2024, 19:55] [ApiHandler] server responded: 403
```

The 403 is expected behaviour in this case, if no X-Client-Id and X-Client-Secret headers are supplied. If the values were to be incorrect instead of missing, it should return HTTP Status 401. This is a slight difference between production (Cloudflare) and test environment (reverse proxy with WireShark packet logging), as Cloudflare uses the "CF-Access-Client-Id" and "CF-Access-Client-Secret" headers instead. But behaviour is identical as it also doesn't send the custom headers there.

**Comment by LeoKlaus** [comment]
Hey Marc, thanks for the report.

I'll take a look at this and report back shortly.

**Comment by LeoKlaus** [comment]
I've implemented a fix. Once the build is approved, you should be able to test it using TestFlight.
I'll let you know once the update is approved.

**Comment by MerelSollie** [comment]
Great to hear, thanks for the quick response!

**Comment by LeoKlaus** [comment]
The update is now live in TestFlight. If you're not part of the beta, you can join using this link https://testflight.apple.com/join/q5WXLtuP

Hope it all works now!

**Comment by MerelSollie** [comment]
I've tested the testflight, and it seems to be working now. Headers are working and detected. Thanks for the quick fix! Is the 2.2.1 build from the testflight planned to hit the App Store soon?

**Comment by LeoKlaus** [comment]
Great, I'm closing this then. I still have to wrap some things up before publishing but it shouldn't take much longer than a week.

**Comment by MerelSollie** [comment]
Alright, thanks!
